src/main/java/com/philips/swcoe/cerberus/cerebellum/codemetrics/java/CodeMetricsHorizontalWriterService.java

```diff
@@ -15,9 +15,9 @@
 public class CodeMetricsHorizontalWriterService extends AbstractCodeMetricsWriterService  {
     private static Logger log = Logger.getLogger(CodeMetricsHorizontalWriterService.class);
 
-    public CodeMetricsHorizontalWriterService(List<String> classConfig, List<String> methodConfig, char delimiter) throws IOException {
-        super(classConfig, methodConfig, delimiter);
-        csvPrinter = getCsvPrinterHorizontal(this.delimiter, reportWriter);
+    public CodeMetricsHorizontalWriterService(List<String> classConfig, List<String> methodConfig, String reportFormat) throws IOException {
```

Comment:
Thats a nice idea, but Generator is too different from writer. Writer is common for all formats of data and generator is not common and we have enum since we have different way to generate report for each format